src/solvers/refinement/string_constraint.h

```diff
@@ -0,0 +1,180 @@
+/*******************************************************************\
+
+Module: String constraints.
+        These are formulas talking about strings. We implemented two
+        forms of constraints: `string_constraintt` implements formulas
+        of the form $\forall univ_var \in [lb,ub[. premise => body$,
+	and not_contains_constraintt implements those of the form:
+        $\forall x in [lb,ub[. p(x) => \exists y in [lb,ub[.
+	s1[x+y] != s2[y]$.
+
+Author: Romain Brenguier, romain.brenguier@diffblue.com
+
+\*******************************************************************/
+
+#ifndef CPROVER_SOLVERS_REFINEMENT_STRING_CONSTRAINT_H
+#define CPROVER_SOLVERS_REFINEMENT_STRING_CONSTRAINT_H
+
+#include <langapi/language_ui.h>
+#include <solvers/refinement/bv_refinement.h>
+#include <solvers/refinement/refined_string_type.h>
+
+class string_constraintt: public exprt
+{
+public:
+  // String constraints are of the form
+  // forall univ_var in [lower_bound,upper_bound[. premise => body
```

Comment:
Actually no, we want to exclude the upper bound, we very often use this kind of constraints with `lower_bound=0` and `upper_bound=s.length` and we want the variable to go from `0` to `s.length-1` because these are the indexes where the array is defined. 
To be less ambiguous we could change the parameters name to be `start` and `length` and the interval would be `[start, start+length[`